marklogic-data-hub/src/test/ml-modules/root/test/suites/data-hub/5/smart-mastering/helper/property-name-to-query.xqy

```diff
@@ -0,0 +1,99 @@
+xquery version "1.0-ml";
+
+import module namespace helper = "http://marklogic.com/smart-mastering/helper-impl"
+  at "/com.marklogic.smart-mastering/matcher-impl/helper-impl.xqy";
+
+
+import module namespace test = "http://marklogic.com/test" at "/test/test-helper.xqy";
+
+
+declare variable $match-options := xdmp:unquote('{
+  "dataFormat": "json",
+  "targetEntityType": "Customer",
+  "propertyDefs": [
+    {
+      "property": {
+        "name": "nicknames",
+        "indexReferences": [
```

Comment:
Is this something that Pari would provide - i.e. the entity model wouldn't specify an index on nicknames? Also, does the GUI support this, or is this something Pari would provide via editing the JSON file? If the GUI doesn't support it (which seems fine), is there a GUI test to ensure it won't drop this config?

Reply:
This is something that Pari would provide. I'll add a Hub Central test to ensure it isn't dropped.

Reply:
@rjrudin Reviewing it further, I believe the test here should be sufficient: https://github.com/marklogic/marklogic-data-hub/blob/1135085a67d780142ae4234003776fd17857f591/marklogic-data-hub-central/src/test/java/com/marklogic/hub/central/controllers/steps/AbstractStepControllerTest.java#L47

It verifies that properties that haven't been explicitly sent in an update are retained.